src/v2/Apps/Order/Routes/Payment/index.tsx

```diff
@@ -225,13 +249,22 @@ export const PaymentRoute: FC<Props> = props => {
       <TwoColumnLayout
         Content={
           <>
-            {isSettingPayment ? (
+            {isSettingPayment &&
+            !isPaymentSetupComplete &&
+            !shouldPollAccountBalance ? (
               <SetPaymentByStripeIntent
                 order={order}
                 setupIntentId={setupIntentId!}
                 onSuccess={onSetPaymentSuccess}
                 onError={onSetPaymentError}
               />
+            ) : shouldPollAccountBalance && setupIntentId ? (
```

Comment:
Given the account balance polling step heavily depends on data and banks in production and we'll be testing and learning, I suggest we implement a flag (can be directly in code or an Unleash flag) that we can turn on/off the account balance polling step easily in case we need it. So

- when account balance polling is on, the flow would be
```
PaymentContent -> SetPaymentByStripeIntent -> PollAccountBalanceQueryRenderer -> Review
```

- when account balance polling is off, the flow would be
```
PaymentContent -> SetPaymentByStripeIntent -> Review
```

Reply:
thank you for this suggestion, makes sense to me 👍